# HG changeset patch
# User Abhinav Tripathi <genuinelucifer@gmail.com>
# Date 1478586984 28800
#      Mon Nov 07 22:36:24 2016 -0800
# Node ID bd8ea9e44230723cabfb77767b073590f4181111
# Parent  7c7a61c2e0edf0ad9249e1c48cb3b9759db0d5be
Allow changing comment character when using Ctrl+R... (bug #49182)

* file-editor-tab.cc (file_editor_tab::comment_string): Get the comment
character to be used from settings.
* settings-dialog.cc (settings_dialog::settings_dialog): Set the previously
selected comment character when the dialog opens.
(settings_dialog::write_changed_settings): Store the settings for comment
character.
* settings-dialog.ui: Add a new layout for the user to select the required
comment character.

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -770,17 +770,26 @@ file_editor_tab::save_apis_info ()
 {
   _lexer_apis->savePrepared (_prep_apis_file);
 }
 
 QString
 file_editor_tab::comment_string (const QString& lexer)
 {
   if (lexer == "octave" || lexer == "matlab")
-    return QString("%");
+    {
+      QSettings *settings = resource_manager::get_settings ();
+      int comment_index = settings->value ("editor/octave_comment_string", 0).toInt ();
+      if (comment_index == 1)
+      	return QString("#");
+      else if (comment_index == 2)
+        return QString("%");
+      else
+        return QString("##");  // default and for index 0
+    }
   else if (lexer == "perl" || lexer == "bash" || lexer == "diff")
     return QString("#");
   else if (lexer == "cpp")
     return ("//");
   else if (lexer == "batch")
     return ("REM ");
   else
     return ("%");  // should never happen
diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -433,16 +433,18 @@ settings_dialog::settings_dialog (QWidge
   ui->editor_ws_checkbox->setChecked (
     settings->value ("editor/show_white_space", false).toBool ());
   ui->editor_ws_indent_checkbox->setChecked (
     settings->value ("editor/show_white_space_indent",false).toBool ());
   ui->cb_show_eol->setChecked (
     settings->value ("editor/show_eol_chars",false).toBool ());
   ui->cb_show_hscrollbar->setChecked (
     settings->value ("editor/show_hscroll_bar",true).toBool ());
+  ui->combo_oct_comment_str->setCurrentIndex (
+    settings->value ("editor/octave_comment_string", 0).toInt ());
 
 #if defined (HAVE_QSCINTILLA)
 #  if defined (Q_OS_WIN32)
   int eol_mode = QsciScintilla::EolWindows;
 #elif defined (Q_OS_MAC)
   int eol_mode = QsciScintilla::EolMac;
 #else
   int eol_mode = QsciScintilla::EolUnix;
@@ -811,16 +813,18 @@ settings_dialog::write_changed_settings 
   settings->setValue ("editor/show_white_space_indent",
                       ui->editor_ws_indent_checkbox->isChecked ());
   settings->setValue ("editor/show_eol_chars",
                       ui->cb_show_eol->isChecked ());
   settings->setValue ("editor/show_hscroll_bar",
                       ui->cb_show_hscrollbar->isChecked ());
   settings->setValue ("editor/default_eol_mode",
                       ui->combo_eol_mode->currentIndex ());
+  settings->setValue ("editor/octave_comment_string",
+                      ui->combo_oct_comment_str->currentIndex ());
   settings->setValue ("editor/default_encoding",
                       ui->editor_combo_encoding->currentText ());
   settings->setValue ("editor/auto_indent",
                       ui->editor_auto_ind_checkbox->isChecked ());
   settings->setValue ("editor/tab_indents_line",
                       ui->editor_tab_ind_checkbox->isChecked ());
   settings->setValue ("editor/backspace_unindents_line",
                       ui->editor_bs_unind_checkbox->isChecked ());
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -912,16 +912,74 @@
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
+              <item>
+               <layout class="QHBoxLayout" name="horizontalLayout_17">
+                <property name="topMargin">
+                 <number>0</number>
+                </property>
+                <item>
+                 <widget class="QLabel" name="label_23">
+                  <property name="text">
+                   <string>Character used to comment selected text (Ctrl+R)</string>
+                  </property>
+                 </widget>
+                </item>
+                <item>
+                 <widget class="QComboBox" name="combo_oct_comment_str">
+                  <property name="sizePolicy">
+                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+                    <horstretch>0</horstretch>
+                    <verstretch>0</verstretch>
+                   </sizepolicy>
+                  </property>
+                  <property name="sizeAdjustPolicy">
+                   <enum>QComboBox::AdjustToContents</enum>
+                  </property>
+                  <property name="minimumContentsLength">
+                   <number>3</number>
+                  </property>
+                  <item>
+                   <property name="text">
+                    <string>##</string>
+                   </property>
+                  </item>
+                  <item>
+                   <property name="text">
+                    <string>#</string>
+                   </property>
+                  </item>
+                  <item>
+                   <property name="text">
+                    <string>%</string>
+                   </property>
+                  </item>
+                 </widget>
+                </item>
+                <item>
+                 <spacer name="horizontalSpacer_33">
+                  <property name="orientation">
+                   <enum>Qt::Horizontal</enum>
+                  </property>
+                  <property name="sizeHint" stdset="0">
+                   <size>
+                    <width>40</width>
+                    <height>20</height>
+                   </size>
+                  </property>
+                 </spacer>
+                </item>
+               </layout>
+              </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_6">
              <property name="title">
               <string>Indentation</string>
              </property>
